### ISSUE TITLE
City fountain check doesn't give any item

### ISSUE THREAD

**Issue by kikones34** [Korzer420/LoreRandomizer#3]
I was playing a rando with some friends and we got softlocked because the City Fountain check has Right Mothwing Cloak, but whenever we check it, it just shows the vanilla text and doesn't give any item. We're not sure if it's a bug or if we're not understanding how the check is supposed to be activated.

**Comment by Korzer420** [comment]
Sorry for the late response, kinda checked it and forgot to write here again :( Can't reproduce this issue. If you stumble on this once more, can you provide your modlog?

**Comment by kikones34** [comment]
Hey! I still have the modlogs laying around, I've looked at them a bit and there's a suspicious exception from ItemChanger:

```
[ERROR]:[ItemChangerMod] - Error invoking ModifyItem for item Right_Mothwing_Cloak with placement unavailable:
[ERROR]:[ItemChangerMod] - System.NullReferenceException: Object reference not set to an instance of an object
[ERROR]:[ItemChangerMod] -   at GrubNames.GrubNames.<Initialize>b__7_0 (ItemChanger.GiveEventArgs args) [0x00019] in <5e26f202e42d4788a86ca390c786597c>:0 
[ERROR]:[ItemChangerMod] -   at (wrapper delegate-invoke) System.Action`1[ItemChanger.GiveEventArgs].invoke_void_T(ItemChanger.GiveEventArgs)
[ERROR]:[ItemChangerMod] -   at ItemChanger.AbstractItem.ModifyItemInvoke (ItemChanger.GiveEventArgs args) [0x00000] in C:\dev\ItemChanger\ItemChanger\AbstractItem.cs:294 
```

I'm not sure why GrubNames would appear in the traceback, not sure if it's related in any way to the issue (I was playing with that mod enabled), but it'd be weird since the item is not even a grub.

This is the only related exception I found, Lore Randomizer didn't log any.

**Comment by Korzer420** [comment]
Hm, can you post your entire modlog here? I suspect some other mod may cause an issue here by also modifying this. Also, since another person has also mentioned this in addition to the fountain, (in case you remember) could you pick up the item at Traitors Grave?

**Comment by kikones34** [comment]
Here's the entire modlog
[ModLog 09 06 2023 (22 00 29).txt](https://github.com/Korzer420/LoreRandomizer/files/14173319/ModLog.09.06.2023.22.00.29.txt)

> could you pick up the item at Traitors Grave

You mean pick up the item in-game? As far as I remember I didn't have access to that zone yet, and were softlocked due to not being able to pick up right mothwing cloak 😓

**Comment by Korzer420** [comment]
Thanks, I'll try to investigate that, although I'm not certain that I can fix that. Likely this is caused by some interaction with another unknown mod. :/

**Comment by Korzer420** [comment]
@kikones34 Hey, I think I found the problem while playing my own rando run. Did you have by any chance "Cursed Listening" activate? Even though it shouldn't, the listening ability did prevent the item give process. In addition to that, the modification doesn't place a "You can't listen/read this" in the interaction which is why it behaves like vanilla.

So two bugs covering each other. I'll fix that in the next few days.

**Comment by Korzer420** [comment]
Fixed in LoreRando 0.3 (+ LoreCore 0.6)

**Comment by kikones34** [comment]
> @kikones34 Hey, I think I found the problem while playing my own rando run. Did you have by any chance "Cursed Listening" activate? Even though it shouldn't, the listening ability did prevent the item give process. In addition to that, the modification doesn't place a "You can't listen/read this" in the interaction which is why it behaves like vanilla.
> 
> So two bugs covering each other. I'll fix that in the next few days.

Yes, we did have "Cursed Listening" active! Thanks for fixing it ❤️ !
